Composer/packages/client/src/components/ImportModal/ImportModal.tsx

```diff
@@ -0,0 +1,338 @@
+  description?: string;
+  eTag: string;
+  name?: string;
+  source: string;
+  templateDir: string;
+  urlSuffix?: string;
+};
+
+type ImportPayload = {
+  name: string;
+  description?: string;
+};
+
+type ExistingProjectInfo = {
+  id: string;
+  location: string;
+  name: string;
+};
+
+type ImportModalState =
+  | 'copyingContent' // after selecting import to existing project
+  | 'connecting' // before login prompt shows
+  | 'downloadingContent' // after logging in
+  | 'failed' // error
+  | 'promptingToSave' // when an existing project is found under the same alias
+  | 'signingIn'; // when login prompt is showing
+
+const boldText = css`
+  font-weight: ${FontWeights.semibold};
+  color: #106ebe;
+  word-break: break-work;
+`;
+
+const CONNECTING_STATUS_DISPLAY_TIME = 2000;
+
+export const ImportModal: React.FC<RouteComponentProps> = (props) => {
+  const { location } = props;
+  const [importSource, setImportSource] = useState<string>('');
+  const [importPayload, setImportPayload] = useState<ImportPayload>({ name: '', description: '' });
+  const [importedProjectInfo, setImportedProjectInfo] = useState<ImportedProjectInfo | undefined>(undefined);
+  const [modalState, setModalState] = useState<ImportModalState>('connecting');
+  const [existingProject, setExistingProject] = useState<ExistingProjectInfo | undefined>(undefined);
+  const [error, setError] = useState<any>(undefined);
+  const [backupLocation, setBackupLocation] = useState<string>('');
+  const { addNotification } = useRecoilValue(dispatcherState);
+
+  const importAsNewProject = useCallback((info: ImportedProjectInfo) => {
+    // navigate to creation flow with template selected
+    const { alias, description, eTag, name, source, templateDir, urlSuffix } = info;
+    let creationUrl = `/projects/create/${encodeURIComponent(source)}?imported=true&templateDir=${encodeURIComponent(
+      templateDir
+    )}&eTag=${encodeURIComponent(eTag)}`;
+
+    if (name) {
+      creationUrl += `&name=${encodeURIComponent(name)}`;
+    }
+    if (description) {
+      creationUrl += `&description=${encodeURIComponent(description)}`;
+    }
+    if (urlSuffix) {
+      creationUrl += `&urlSuffix=${encodeURIComponent(urlSuffix)}`;
+    }
+    if (alias) {
+      creationUrl += `&alias=${encodeURIComponent(alias)}`;
+    }
+
+    navigate(creationUrl);
+  }, []);
+
+  const importToExistingProject = useCallback(() => {
+    if (importedProjectInfo && existingProject) {
+      setModalState('copyingContent');
+
+      const backupAndSave = async () => {
+        try {
+          // call server to do backup and then save to existing project
+          let res = await fetch(`/api/projects/${existingProject.id}/backup`, { method: 'POST' });
+          const { path } = await res.json();
+          setBackupLocation(path);
+
+          const { eTag, templateDir } = importedProjectInfo;
+          res = await fetch(`/api/projects/${existingProject.id}/copyTemplateToExisting`, {
```

Comment:
is there a particular reason that you are using the vanilla fetch api instead of axios (httpClient)?

Reply:
Nope, I can switch over to Axios